lua/nvim-tree/explorer/init.lua

```diff
@@ -683,6 +686,102 @@ function Explorer:get_nodes()
   return self:clone()
 end
 
+---@param new_tabpage integer
+---@return boolean
+function Explorer:is_window_event(new_tabpage)
+  local is_event_scope_window = vim.v.event.scope == "window" or vim.v.event.changed_window or false
+  return is_event_scope_window and new_tabpage == self.current_tab
+end
+
+---@param name string
+---@return string|nil
+function Explorer:clean_input_cwd(name)
+  name = vim.fn.fnameescape(name)
+  local cwd = core.get_cwd()
+  if cwd == nil then
+    return
+  end
+  local root_parent_cwd = vim.fn.fnamemodify(utils.path_remove_trailing(cwd), ":h")
+  if name == ".." and root_parent_cwd then
+    return vim.fn.expand(root_parent_cwd)
+  else
+    return vim.fn.expand(name)
+  end
+end
+
+---@param foldername string
+---@return boolean
+function Explorer:prevent_cwd_change(foldername)
+  local is_same_cwd = foldername == core.get_cwd()
+  local is_restricted_above = config.restrict_above_cwd and foldername < vim.fn.getcwd(-1, -1)
+  return is_same_cwd or is_restricted_above
+end
+
+---@param f function
+---@return fun(foldername: string, should_open_view: boolean|nil)
+function Explorer:add_profiling_to(f)
+  return function(foldername, should_open_view)
+    local profile = log.profile_start("change dir %s", foldername)
+    f(foldername, should_open_view)
+    log.profile_end(profile)
+  end
+end
+
+---@return boolean
+function Explorer:should_change_dir()
+  return config.enable and vim.tbl_isempty(vim.v.event)
+end
+
+---@param global boolean
+---@param path string
+function Explorer:cd(global, path)
+  vim.cmd((global and "cd " or "lcd ") .. vim.fn.fnameescape(path))
+end
+
+function Explorer:force_dirchange(folder_name, with_open)
+  local fn = self:add_profiling_to(function(foldername, should_open_view)
+    local valid_dir = vim.fn.isdirectory(foldername) == 1 -- prevent problems on non existing dirs
+    if valid_dir then
+      if self:should_change_dir() then
+        self:cd(config.global, foldername)
+      end
+      core.init(foldername)
+    end
+
+    if should_open_view then
+      require("nvim-tree.lib").open()
+    else
+      local explorer = core.get_explorer()
+      if explorer then
+        explorer.renderer:draw()
+      end
```

Comment:
We already have the explorer - `self` - so can just call `self.renderer:draw()` directly.